tests/samplers_tests/test_nsgaii.py

```diff
@@ -336,266 +338,111 @@ def test_constrained_dominates_infeasible_vs_infeasible(direction: StudyDirectio
-    ]
-    population_per_rank = sampler._fast_non_dominated_sort(trials, directions)
-    assert [{t.number for t in population} for population in population_per_rank] == [
-        {2},
-        {0, 3, 5},
-        {1},
-        {4},
-    ]
-
+def _check_non_dominated_sort(
+    trials: List[FrozenTrial],
+    direction: List[StudyDirection],
+    population_per_rank: List[List[FrozenTrial]],
+) -> None:
+    # Check that the number of trials do not change.
+    flattened = [trial for rank in population_per_rank for trial in rank]
+    assert len(flattened) == len(trials)
 
-def test_fast_non_dominated_sort_constrained_feasible() -> None:
     with warnings.catch_warnings():
-        warnings.simplefilter("ignore", optuna.exceptions.ExperimentalWarning)
-        sampler = NSGAIISampler(constraints_func=lambda _: [0])
+        warnings.simplefilter("ignore", UserWarning)
+        # Check that the trials in the same rank do not dominate each other.
+        for i in range(len(population_per_rank)):
+            for trial1 in population_per_rank[i]:
+                for trial2 in population_per_rank[i]:
 
-    # Single objective.
-    directions = [StudyDirection.MINIMIZE]
-    trials = [
-        _create_frozen_trial(0, [10], [0]),
-        _create_frozen_trial(1, [20], [0]),
-        _create_frozen_trial(2, [20], [0]),
-        _create_frozen_trial(3, [30], [0]),
-    ]
-    population_per_rank = sampler._fast_non_dominated_sort(trials, directions)
-    assert [{t.number for t in population} for population in population_per_rank] == [
-        {0},
-        {1, 2},
-        {3},
-    ]
-
-    # Two objective.
-    directions = [StudyDirection.MAXIMIZE, StudyDirection.MAXIMIZE]
-    trials = [
-        _create_frozen_trial(0, [10, 30], [0]),
-        _create_frozen_trial(1, [10, 10], [0]),
-        _create_frozen_trial(2, [20, 20], [0]),
-        _create_frozen_trial(3, [30, 10], [0]),
-        _create_frozen_trial(4, [15, 15], [0]),
-    ]
-    population_per_rank = sampler._fast_non_dominated_sort(trials, directions)
-    assert [{t.number for t in population} for population in population_per_rank] == [
-        {0, 2, 3},
-        {4},
-        {1},
-    ]
-
-    # Three objective.
-    directions = [StudyDirection.MAXIMIZE, StudyDirection.MAXIMIZE, StudyDirection.MINIMIZE]
-    trials = [
-        _create_frozen_trial(0, [5, 5, 4], [0]),
-        _create_frozen_trial(1, [5, 5, 5], [0]),
-        _create_frozen_trial(2, [9, 9, 0], [0]),
-        _create_frozen_trial(3, [5, 7, 5], [0]),
-        _create_frozen_trial(4, [0, 0, 9], [0]),
-        _create_frozen_trial(5, [0, 9, 9], [0]),
-    ]
-    population_per_rank = sampler._fast_non_dominated_sort(trials, directions)
-    assert [{t.number for t in population} for population in population_per_rank] == [
-        {2},
-        {0, 3, 5},
-        {1},
-        {4},
-    ]
+                    assert not _constrained_dominates(trial1, trial2, direction)
```

Comment:
I'm wondering if these lines can check any behavior of `_non_dominated_sort` since `trial1` and `trial2` are completely the same `FrozenTrial` objects. What do you think?